layers/+misc/tree-sitter/packages.el

```diff
@@ -0,0 +1,100 @@
+;;; packages.el --- tree-sitter layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Elliott Shugerman <eeshugerman@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+(defconst tree-sitter-packages
+  '(tree-sitter
+    tree-sitter-langs
+    (tree-sitter-indent
+     :toggle tree-sitter-indent-enable)
+    (ts-fold
+     :toggle tree-sitter-fold-enable
+     :location (recipe
+                :fetcher github
+                :repo "jcs090218/ts-fold"))))
+
+(defun tree-sitter/init-tree-sitter ()
+  (use-package tree-sitter
+    :defer t
+    :init
+    (progn
+      (when tree-sitter-syntax-highlight-enable
+       (add-hook 'tree-sitter-after-on-hook #'tree-sitter-hl-mode)))
+    :config
+    (progn
+      (global-tree-sitter-mode))))
+
+(defun tree-sitter/init-tree-sitter-langs ()
+  (use-package tree-sitter-langs
+    :defer t))
+
+(defun tree-sitter/init-tree-sitter-indent ()
+  (use-package tree-sitter-indent
+    :if tree-sitter-indent-enable
+    :defer t
+    :init
+    (progn
+      (add-hook 'rust-mode-hook #'tree-sitter-indent-mode))))
+
+(defconst tree-sitter//ts-fold-supported-major-mode-hooks
```

Comment:
Would be nice if there were a way to avoid listing these out explicitly and still defer loading. There's `global-ts-fold-mode` but it breaks folding for unsupported major modes. Would it make sense to ask the developer to accommodate this, either in `global-ts-fold-mode` or by autoloading something like `ts-fold-supported-modes`?

In other words I'm not sure how this usually works. If this is the way to do it then it works for me but if there's a better way to do it I'm up for it.

Reply:
Hmmmm I think having a list of supported modes in the emacs package would be the way to go. But thats just an improvement, for the first version this is ok.